mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -103,11 +103,30 @@ impl S3CrtClient {
         retry_strategy_options.backoff_retry_options.jitter_mode = ExponentialBackoffJitterMode::Full;
         let retry_strategy = RetryStrategy::standard(&allocator, &retry_strategy_options).unwrap();
 
-        let mut client_config = ClientConfig::new();
+        let credentials_provider = match config.profile_name_override {
```

Comment:
In the previous revision I thought we didn’t construct the credentials provider at all if `no_sign_request` was set — why’d we change that?

Reply:
`credentials_provider` will be dropped if put it into
```
if !config.no_sign_request {
.....
}
```

and I thought we don't want to [singing_config](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3-crt/src/auth/signing_config.rs#L11-L20)owns it and let crt copy/increase counts

Reply:
Not sure what you mean. `init_default_signing_config` takes a `&CredentialsProvider` as input, so we know that it's OK to drop the `CredentialsProvider` after the call to `init_default_signing_config`. Otherwise Rust wouldn't let this code compile.

Reply:
Actually, unless I'm missing something, I think the existing interface is just wrong. It calls `aws_s3_init_default_signing_config`, which doesn't seem to increment the reference count on the credentials provider it's given.

I think we need to change `init_default_signing_config` to take ownership of the `CredentialsProvider`, and then add that as a field to `SigningConfigInner` so that it lives as long as the underlying signing config does.

Reply:
> I think we need to change init_default_signing_config to take ownership of the CredentialsProvider, and then add that as a field to SigningConfigInner so that it lives as long as the underlying signing config does.

This is the last revision, seems like my understanding of your last comment was wrong.

Reply:
> so we know that it's OK to drop the CredentialsProvider after the call to init_default_signing_config.
rust would compile, but when it will result segmentation fault.

Reply:
side notes, let's reach consensus on Pull Request next time to get rid of these back and forth